### PR TITLE
Dev/saars/revisit hosting startup example

### PR DESCRIPTION
**Pull request by xiaomi7732**
Part of #265.

### CODE REVIEW

File: examples/ZeroUserCodeLightup.Net6/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+All you need to do is to add a few lines in the [Dockerfile](./dockerfile), and they are:
+
+```dockerfile
+...
+# Adding a reference to hosting startup package
+RUN dotnet add package Microsoft.ApplicationInsights.Kubernetes.HostingStartup --prerelease
+
+# Light up Application Insights for Kubernetes
+ENV APPINSIGHTS_INSTRUMENTATIONKEY $APPINSIGHTS_KEY
```
Comment (karolz-ms) [inline examples/ZeroUserCodeLightup.Net6/README.md:15]: We should not direct the user to embed the instrumentation key into the image, even for testing purposes.

File: examples/ZeroUserCodeLightup.Net6/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ENV ASPNETCORE_HOSTINGSTARTUPASSEMBLIES Microsoft.ApplicationInsights.Kubernetes.HostingStartup
```
Comment (karolz-ms) [inline examples/ZeroUserCodeLightup.Net6/README.md:16]:
```suggestion
ENV ASPNETCORE_HOSTINGSTARTUPASSEMBLIES=Microsoft.ApplicationInsights.Kubernetes.HostingStartup
```
Omitting the equal sign [is not recommended by Docker](https://docs.docker.com/engine/reference/builder/#env)

File: examples/ZeroUserCodeLightup.Net6/dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# Create an argument to allow docker builder to passing in application insights key.
+# For example: docker build . --build-arg APPINSIGHTS_KEY=YOUR_APPLICATIONINSIGHTS_INSTRUMENTATION_KEY
+ARG APPINSIGHTS_KEY
+# Making sure the argument is set. Fail the build of the container otherwise.
+RUN test -n "$APPINSIGHTS_KEY"
+
+# Light up Application Insights for Kubernetes
+ENV APPINSIGHTS_INSTRUMENTATIONKEY $APPINSIGHTS_KEY
+ENV ASPNETCORE_HOSTINGSTARTUPASSEMBLIES Microsoft.ApplicationInsights.Kubernetes.HostingStartup
```
Comment (karolz-ms) [inline examples/ZeroUserCodeLightup.Net6/dockerfile:22]:
```suggestion
ENV ASPNETCORE_HOSTINGSTARTUPASSEMBLIES=Microsoft.ApplicationInsights.Kubernetes.HostingStartup
```

File: examples/ZeroUserCodeLightup.Net6/dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+COPY . ./
+# Adding a reference to hosting startup package
+# Remove '--prerelease' if you want to stay only on stable version.
+RUN dotnet add package Microsoft.ApplicationInsights.Kubernetes.HostingStartup --prerelease
+RUN dotnet publish -c Release -o out
+
+# Build runtime image
+FROM mcr.microsoft.com/dotnet/aspnet:6.0
+
+# Create an argument to allow docker builder to passing in application insights key.
```
Comment (karolz-ms) [inline examples/ZeroUserCodeLightup.Net6/dockerfile:14]: Let's not do that. Instead, explain to the user how to ensure that `APPINSIGHTS_INSTRUMENTATIONKEY` is set when running the image locally using Docker, as well as when running on Kubernetes.

**Review by karolz-ms — CHANGES_REQUESTED** [review]

File: examples/ZeroUserCodeLightup.Net6/README.md (hunk 2 above)
Comment (xiaomi7732) [reply examples/ZeroUserCodeLightup.Net6/README.md]: #TIL and fixed.

File: examples/ZeroUserCodeLightup.Net6/dockerfile (hunk 3 above)
Comment (xiaomi7732) [reply examples/ZeroUserCodeLightup.Net6/dockerfile]: Fixed.

File: examples/ZeroUserCodeLightup.Net6/dockerfile (hunk 4 above)
Comment (xiaomi7732) [reply examples/ZeroUserCodeLightup.Net6/dockerfile]: Done.

**Review by karolz-ms — APPROVED** [review]
Super, thanks

**Pull request merged**